plugins/inputs/socket_listener/socket_listener_test.go

```diff
@@ -178,6 +185,158 @@ func TestSocketListener(t *testing.T) {
 	}
 }
 
+func TestCases(t *testing.T) {
+	// Get all directories in testdata
+	folders, err := os.ReadDir("testcases")
+	require.NoError(t, err)
+
+	// Register the plugin
+	inputs.Add("socket_listener", func() telegraf.Input {
+		return &SocketListener{}
+	})
+
+	// Prepare the influx parser for expectations
+	parser := &influx.Parser{}
+	require.NoError(t, parser.Init())
+
+	for _, f := range folders {
```

Comment:
It might be useful to run each folder in a separate t.Run() so you can give it a name in test output

Reply:
Then again, t.Run() would let the tests run in parallel which wouldn't work since they share the same parser. We would have to move the parser initialization inside the t.Run()

Reply:
Ok.